cli/azd/cmd/config.go

```diff
@@ -19,16 +19,20 @@ func configCmd(rootOptions *internal.GlobalCommandOptions) *cobra.Command {
 	root := &cobra.Command{
 		Use:   "config",
 		Short: "Manage Azure Developer CLI configuration",
+		Long: `Manage the Azure Developer CLI user configuration, which includes your default Azure subscription and location.
+
+The easiest way to initially configure azd is to run ` + output.WithBackticks("azd init") + `.
+The subscription and location you select will be stored in the config.json file located at $AZURE_CONFIG_DIR.
+The default value of AZURE_CONFIG_DIR is $HOME/.azd on Linux and macOS, and %USERPROFILE%\.azd on Windows.`,
```

Comment:
We could have a function to get the path depending on the OS.  That could be a better user experience.
All messages from below would be able to mention the actual path without referencing `AZURE_CONFIG_DIR`

I see a potential confusion when we introduce `AZURE_CONFIG_DIR`, as it looks like an env var that you can set or that it is already defined in the env and you can inspect.  But we are only using it as an indirect reference for the documentation cross OS.

Not a blocking thing for the PR, but I would consider improving the experience to user, as we know when we are running on Windows/Lin/Mac

Reply:
Good idea - I was able to implement this easily and fall back to `$AZURE_CONFIG_DIR` in the event there is an error getting the user config dir.